File: src/Domain/Response.java
package Domain;

public class Response<T> {
    private T value;
    private boolean wasException=false;
    private String message;

    public Response(Exception exception, String message){
        wasException=true;
        this.message=message;
    }

    public Response(T value, String message){
        this.value=value;
        this.message=message;
    }

    public T getValue() {
        return value;
    }

    public boolean WasException() {
        return wasException;
    }

    public String getMessage() {
        return message;
    }
}

File: Client/src/API/Response.js



export class Response {
    constructor(response) {
        this.value = response.value;
        this.message = response.message;
        this.was_exception = response.wasException;
    }
    static create(value, message, was_exception) {
        return new Response({
            value: value,
            message: message,
            was_exception: was_exception,

        })

    }
}